src/moscot/problems/cross_modality/_mixins.py

```diff
@@ -0,0 +1,165 @@
+from typing import TYPE_CHECKING, Any, Literal, Optional
+
+import pandas as pd
+
+from anndata import AnnData
+
+from moscot import _constants
+from moscot._types import ArrayLike, Str_Dict_t
+from moscot.base.problems._mixins import AnalysisMixin, AnalysisMixinProtocol
+from moscot.base.problems.compound_problem import B, K
+
+__all__ = ["CrossModalityTranslationMixin"]
+
+
+class CrossModalityTranslationMixinProtocol(AnalysisMixinProtocol[K, B]):
+    """Protocol class."""
+
+    adata_src: AnnData
+    adata_tgt: AnnData
+    _src_attr: Optional[str]
+    _tgt_attr: Optional[str]
+    _translation: Optional[ArrayLike]
+    batch_key: Optional[str]
+
+    def _cell_transition(self: AnalysisMixinProtocol[K, B], *args: Any, **kwargs: Any) -> pd.DataFrame:
+        ...
+
+
+class CrossModalityTranslationMixin(AnalysisMixin[K, B]):
+    """Cross modality translation analysis mixin class."""
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self._src_attr: Optional[str] = None
+        self._tgt_attr: Optional[str] = None
+
+    def translate(  # type: ignore[misc]
+        self: CrossModalityTranslationMixinProtocol[K, B],
+        source: K,
+        target: K,
+        forward: bool = True,
+        overwrite: bool = False,
+        **kwargs: Any,
+    ) -> ArrayLike:
+        """
+        Translate source or target object.
+
+        Parameters
+        ----------
+        source
+            Key identifying the source distribution.
```

Comment:
Actually, this is not about the key in the adata object where the data is stored. Here I mean the key related to policy, i.e. (src, tgt) in case of dummy policy or (i, "ref") in case of external star policy. So the key added to the problem after calling hte prepare function - don't know how this is called/ described exactly?

Reply:
`Key defining the batches in the source distribution.`